src/api/app/models/package.rb

```diff
@@ -1379,6 +1379,15 @@ def file_exists?(filename)
   def has_icon?
     file_exists?("_icon")
   end
+
+  def what_depends_on(repository, architecture)
+    begin
+      path = "/build/#{project.name}/#{repository}/#{architecture}/_builddepinfo?package=#{name}&view=revpkgnames"
+      [Xmlhash.parse(Suse::Backend.get(path).body).try(:[], 'package').try(:[], 'pkgdep')].flatten.compact
+    rescue ActiveXML::Transport::Error
```

Comment:
I changed it to ``ActiveXML::Transport::NotFoundError``, this is the only thing what can happen atm when you request for wrong repository or architecture.